### PR TITLE
Filesystem PVC as VM disk

### PR DESCRIPTION
**Pull request by davidvossel**
This is built off the work in #693

- Adds filesystem PVC disk support
- Replaces direct ISCSI volume support with new generic PVC volumes
- Revises storage functional tests

### CODE REVIEW

**Comment by cynepco3hahue** [comment]
@davidvossel Can you also adapt tests under `tests/vm_configuration_test.go` from the usage of direct LUN to PVC?

**Comment by davidvossel** [comment]
> Can you also adapt tests under tests/vm_configuration_test.go from the usage of direct LUN to PVC?

@cynepco3hahue Is there any reason to convert those LUN functional tests that exercise the bus?  There's a registry disk test that verifies every driver works.

**Comment by cynepco3hahue** [comment]
@davidvossel I think it no reason :smile: I checked `converter` code and we assign bus only under `Convert_v1_Disk_To_api_Disk` so it does not really matter what volume source do we use.
I think we can just remove them(I already prepared the PR so you do not need to do it under your PR) if @fromanirh does not have any objections.

**Comment by davidvossel** [comment]
@cynepco3hahue Those tests were already removed from this PR. I had to do that to get the tests to compile since i stripped out LUN support.

File: tests/utils.go [hunk 1]
```diff
@@ -360,16 +327,27 @@ func newPVC(os string, withAuth bool) *k8sv1.PersistentVolumeClaim {
 	}
 }
 
-func newPV(os string, lun int32, withAuth bool) *k8sv1.PersistentVolume {
+func createPvISCSI(os string, lun int32, withAuth bool) {
```
Comment (cynepco3hahue) [inline tests/utils.go]: If we drop ISCSI volume source support we can remove all functions related to it

**Comment by cynepco3hahue** [comment]
@davidvossel I see you already removed them

File: tests/utils.go (hunk 1 above)
Comment (davidvossel) [reply tests/utils.go]: This function is creating the PV for the iscsi demo target. All the direct LUN related functions have been removed now.

File: images/iscsi-demo-target-tgtd/run-tgt.sh [hunk 2]
```diff
@@ -40,6 +46,25 @@ if [ -n "$PASSWORD" ] && [ -n "$USERNAME" ]; then
 fi
 
 LUNID=1
+
+convert_to_filesystem() {
+    local FN=$1
+    local DISK_NAME="/tmp/$(basename $FN)"
+    dd if=/dev/zero of=$DISK_NAME bs=1M count=1024
+    mkfs.ext4 $DISK_NAME
+    sleep 10
```
Comment (davidvossel) [inline images/iscsi-demo-target-tgtd/run-tgt.sh]: @cynepco3hahue  do you know why this sleep is here?

Comment (cynepco3hahue) [reply images/iscsi-demo-target-tgtd/run-tgt.sh]: @davidvossel I saw that sometimes mount failed without sleep

**Comment by fabiand** [comment]
Hm, standard ci passed:

```
Ran 36 of 36 Specs in 1461.028 seconds

SUCCESS! -- 36 Passed | 0 Failed | 0 Pending | 0 Skipped PASS
```

It just failed to tear down a node.

**Comment by fabiand** [comment]
ci test please

**Comment by davidvossel** [comment]
> Can't we just look up in the tests where the iscsi pod is deployed and fetch the right IP in the tests? I would really prefer that, instead of switching to the daemonset.

@rmohr the problem is manual testing and development. we can't make a VM manifest that just works (like cluster/vm-pvc.yaml) if someone has to go in and manually enter a POD ip into the manifest after every deploy. During development that's tedious and is error prone. I'm favoring ease of use over efficiency here for the sake of the demo.

**Comment by davidvossel** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

**Comment by davidvossel** [comment]
retest this please

**Comment by davidvossel** [comment]
retest this please

**Comment by davidvossel** [comment]
@rmohr I rebased and the registry-disk failures went away. jenkins passes now, hooray!

**Comment by rmohr** [comment]
> Can't we just look up in the tests where the iscsi pod is deployed and fetch the right IP in the tests? I would really prefer that, instead of switching to the daemonset.

Since it is already done, my biggest concern (more work) is gone. Fine with proceeding with the localhost solution.

File: cluster/vagrant-kubernetes/setup_common.sh [hunk 3]
```diff
@@ -34,7 +34,7 @@ yum -y remove NetworkManager firewalld
 yum -y install https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm
 yum -y install jq sshpass
 
-yum -y install bind-utils net-tools
+yum -y install bind-utils net-tools iscsi-initiator-utils
```
Comment (rmohr) [inline cluster/vagrant-kubernetes/setup_common.sh]: We are now trying to install them two times. Not a blocker since it save to do so.

**Pull request merged**